service/src/main/java/com/codeforcommunity/processor/TeamsProcessorImpl.java

```diff
@@ -35,4 +74,279 @@ public void disbandTeam(JWTData userData, int teamId) {
+    checkTeamExists(teamId);
+    checkGoalExists(goalId);
+
+    UsersTeamsRecord usersTeamsRecord =
+        db.selectFrom(USERS_TEAMS)
+            .where(USERS_TEAMS.USER_ID.eq(userData.getUserId()))
+            .and(USERS_TEAMS.TEAM_ID.eq(teamId))
+            .fetchOne();
+
+    if (usersTeamsRecord != null && usersTeamsRecord.getTeamRole() == TeamRole.LEADER) {
+      db.deleteFrom(GOALS).where(GOALS.ID.eq(goalId));
+    } else {
+      throw new WrongTeamRoleException(teamId, usersTeamsRecord.getTeamRole());
+    }
+  }
+
+  @Override
+  public void inviteUser(JWTData userData, InviteUserRequest inviteUserRequest) {
+    checkTeamExists(inviteUserRequest.getTeamId());
+    UsersTeamsRecord inviterUserTeamsRecord =
+        db.selectFrom(USERS_TEAMS)
+            .where(USERS_TEAMS.USER_ID.eq(userData.getUserId()))
+            .and(USERS_TEAMS.TEAM_ID.eq(inviteUserRequest.getTeamId()))
+            .fetchOne();
+
+    if (inviterUserTeamsRecord != null && inviterUserTeamsRecord.getTeamRole() == TeamRole.LEADER) {
+      inviteUserRequest
+          .getUsers()
+          .forEach(
+              (name, email) -> {
+                if (!db.fetchExists(
+                    USERS_TEAMS
+                        .join(USERS)
+                        .on(USERS.ID.eq(USERS_TEAMS.USER_ID))
+                        .where(
+                            USERS
+                                .EMAIL
+                                .eq(email)
+                                .and(TEAMS.ID.eq(inviteUserRequest.getTeamId()))))) {
+                  UsersRecord invitedUser =
+                      db.selectFrom(USERS).where(USERS.EMAIL.eq(email)).fetchOne();
+                  if (invitedUser != null) {
+                    UsersTeamsRecord usersTeams = db.newRecord(USERS_TEAMS);
+                    usersTeams.setTeamId(inviteUserRequest.getTeamId());
+                    usersTeams.setUserId(invitedUser.getId());
+                    usersTeams.setTeamRole(TeamRole.PENDING);
+                    usersTeams.store();
+                    // TODO SEND EMAIL TO ACCEPT ROLE ON TEAM
+                  } else {
+                    // TODO SEND EMAIL WITH INVITE TO CREATE USER / REQUEST TO JOIN TEAM
+                  }
+                }
+              });
+    } else {
+      throw new WrongTeamRoleException(
+          inviteUserRequest.getTeamId(), inviterUserTeamsRecord.getTeamRole());
+    }
+  }
+
+  @Override
+  public List<UsersTeamDataResponse> getApplicants(JWTData userData, int teamId) {
+    checkTeamExists(teamId);
+    List<UsersTeamsRecord> applicants =
+        db.selectFrom(USERS_TEAMS).where(USERS_TEAMS.TEAM_ROLE.eq(TeamRole.PENDING)).fetch();
+    List<UsersTeamDataResponse> responses = new ArrayList<>();
+    applicants.forEach(
+        app -> {
+          UsersTeamDataResponse response =
+              new UsersTeamDataResponse(app.getUserId(), teamId, app.getTeamRole());
```

Comment:
Based on the API Specification this DTO should just take the `app.getUserId()` and `app.getUsername()`. It doesn't make sense to include teamId and teamRole because they're the same for every applicant

Reply:
Add a TODO for this inline if you're not addressing it here